src/node_trace_events.cc

```diff
@@ -0,0 +1,164 @@
+    return false;
+  }
+}
+
+static void Emit(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  // Args: [type, category, name, id, argName, argValue]
+  CHECK_GE(args.Length(), 3);
+
+  // Check the category group first, to avoid doing more work if it's not
+  // enabled.
+  const char* category_group = GetCategoryGroup(env, args[1]);
+  const uint8_t* category_group_enabled =
+    GetCategoryGroupEnabled(category_group);
+  if (*category_group_enabled == 0) return;
+
+  char phase = GetPhase(env, args[0]);
+  if (phase == 0) return;
+
+  if (!args[2]->IsString()) {
+    env->ThrowTypeError("trace_events name must be a string.");
+    return;
+  }
+  Utf8Value nameValue(env->isolate(), args[2]);
+  const char* name = nameValue.out();
+
+  int64_t id = 0;
+  if (args.Length() >= 4 && !GetId(env, args[3], &id)) return;
+
+  // TODO(@AndreasMadsen): Two extra arguments are not supported.
+  // TODO(@AndreasMadsen): String values are not supported.
+  int32_t num_args = 0;
+  const char* arg_names[1];
+  uint8_t arg_types[1];
+  uint64_t arg_values[1];
+
+  // Create Utf8Value in the function scope to prevent deallocation.
+  // The c-string will be copied by TRACE_EVENT_API_ADD_TRACE_EVENT at the end.
+  Utf8Value arg1NameValue(env->isolate(), args[4]);
+
+  if (args.Length() < 6 || (args[4]->IsUndefined() && args[5]->IsUndefined())) {
+    num_args = 0;
+  } else if (args[4]->IsString() && args[5]->IsNumber()) {
+    num_args = 1;
+    arg_names[0] = arg1NameValue.out();
+    arg_types[0] = TRACE_VALUE_TYPE_INT;
+    arg_values[0] = args[5]->IntegerValue(env->context()).ToChecked();
+  } else if (!args[4]->IsString()) {
+    env->ThrowTypeError("trace_events argName must be string.");
+    return;
+  } else if (!args[5]->IsNumber()) {
+    env->ThrowTypeError("trace_events argValue must be number.");
+    return;
+  }
+
+  // The TRACE_EVENT_FLAG_COPY flag indicates that the name and argument
+  // name should be copied thus they don't need to long-lived pointers.
+  // The category name still won't be copied and thus need to be a long-lived
+  // pointer.
+  uint32_t flags = TRACE_EVENT_FLAG_COPY;
+  if (id != 0) {
+    flags |= TRACE_EVENT_FLAG_HAS_ID;
+  }
+
+  const char* scope = node::tracing::kGlobalScope;
+  uint64_t bind_id = node::tracing::kNoId;
+
+  TRACE_EVENT_API_ADD_TRACE_EVENT(
+    phase, category_group_enabled, name, scope, id, bind_id,
+    num_args, arg_names, arg_types, arg_values,
+    flags);
+}
+
+static void CategoryGroupEnabled(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  const char* category_group = GetCategoryGroup(env, args[0]);
+  const uint8_t* category_group_enabled =
+    GetCategoryGroupEnabled(category_group);
+  if (*category_group_enabled == 0) {
+    args.GetReturnValue().Set(False(env->isolate()));
+  } else {
+    args.GetReturnValue().Set(True(env->isolate()));
+  }
```

Comment:
If I am remembering correctly, I *believe* this could just be...

```c
args.GetReturnValue().Set((*category_group_enabled == 0));
```

That is, I *think* `Set()` accepts `bool` values and handles those appropriately. I could be wrong tho.

Reply:
You are right https://v8docs.nodesource.com/node-8.0/da/da7/classv8_1_1_return_value.html#abbe0803e94a0eb36d04d9bd17ecb96a0

Reply:
Yup – you don’t even need the double parentheses :)